gromacs/config.py

```diff
@@ -639,11 +639,15 @@ def set_gmxrc_environment(gmxrc):
     then only a warning will be logged. Thus, it should be safe to just call
     this function.
     """
-    envvars = ['GMXPREFIX', 'GMXBIN', 'GMXLDLIB', 'GMXMAN', 'GMXDATA',
-               'GROMACS_DIR', 'LD_LIBRARY_PATH', 'MANPATH', 'PKG_CONFIG_PATH',
-               'PATH']
+    # only v5: 'GMXPREFIX', 'GROMACS_DIR'
+    envvars = ['GMXBIN', 'GMXLDLIB', 'GMXMAN', 'GMXDATA',
+               'LD_LIBRARY_PATH', 'MANPATH', 'PKG_CONFIG_PATH',
+               'PATH',
+               'GMXPREFIX', 'GROMACS_DIR']
+    # in order to keep empty values, add ___ sentinels around result
+    # (will be removed later)
     cmdargs = ['bash', '-c', ". {0} && echo {1}".format(gmxrc,
-               ' '.join(['${0}'.format(v) for v in envvars]))]
+               ' '.join(['___${{{0}}}___'.format(v) for v in envvars]))]
```

Comment:
If a env var such as `GMXPREFIX` or `GROMACS_DIR` is not defined then it produces an empty value. Empty values are not picked up when the output is parsed and lead to wrong assignments of values to env vars. That was a problem with Gromacs 4.

Moving unknown env vars to the end would have been sufficient but using "Sentinel" markers around will ensure that even an empty result is picked up because it produces `___'___` (the ' should not be there but indicates where "nothing" is). The line below then removes the sentinels. Cludgy but fairly safe.

(Of course, everything just breaks if the env var values contain spaces as in `GMXBIN=/Users/joe/Stuff for work/gromacs`...)